src/NewsletterStatus/Service/NewsletterStatus.php

```diff
@@ -14,6 +14,7 @@
 use OxidEsales\GraphQL\Account\NewsletterStatus\DataType\NewsletterStatusUnsubscribe as NewsletterStatusUnsubscribeType;
 use OxidEsales\GraphQL\Account\NewsletterStatus\DataType\Subscriber as SubscriberType;
 use OxidEsales\GraphQL\Account\NewsletterStatus\Exception\SubscriberNotFound;
+use OxidEsales\GraphQL\Account\NewsletterStatus\Infrastructure\NewsletterStatus as NewsletterStatusInfrastructure;
 use OxidEsales\GraphQL\Account\NewsletterStatus\Infrastructure\Repository as NewsletterStatusRepository;
```

Comment:
we should rethink the naming, `NewsletterStatusInfrastructure` seems misleading, as it is a Repository. I would suggest to rename the Repostory to just `NewsletterRepository` or maybe better `NewsletterSubscriptionRepository` and then rename `NewsletterStatusInfrastructure` to `NewsletterStatusRepository`